doc/design/model_market.md

```diff
@@ -0,0 +1,142 @@
+
+## Steps to View Model Definitions and Trained Models
+
+<p align="center">
+<img src="figures/list_model_defs.png">
+</p>
+
+<p align="center">
+<img src="figures/list_trained_model.png">
+</p>
+
+1. **Optional**: Login to model market. If the user is not logged in, he can only view public model definitions and trained models.
+1. Click at the "Model Definitions" tab to see the list of model definition Docker images and the model class names in each Docker image.
+1. Click at "Trained Models" tab to see all trained models the current user have published by using SQLFlow `PUBLISH` statement, the evaluation result of the trained model will also be available.
+
+Click at one entry in the list can view the details of the model definition or the trained model:
+
+<p align="center">
+<img src="figures/model_def_detail.png">
+</p>
+
+<p align="center">
+<img src="figures/trained_model_detail.png">
+</p>
+
+## Steps to Publish a Docker Image
+
+<p align="center">
+<img src="figures/new_model_def.png">
+</p>
+
+1. Login to model market.
+1. Go to the "Model Definitions" tab.
+1. Click "New" to add a new image.
+1. Input the full Docker image address in the dialog and click "OK".
+1. ***Optional***: The system will call the Docker registry API to grant access for SQLFlow to pull the image. When using a public image, this step will be skipped. If access can be granted, a message should be shown on the web page.
+1. The system will start to run several checks and tests using the Docker image. If all the checks have passed, the image is added.
+
+**NOTE: Docker image with different [tags](https://www.freecodecamp.org/news/an-introduction-to-docker-tags-9b5395636c2a/) will be recognized as different images.
+
+## Steps to Share Model Definition Docker Images to Other Users
+
+1. Login to model market.
+1. Go to the "Model Definitions" tab.
+1. Find a Docker image in the list to be shared and click on the button "share".
+1. Input the user name or user ID to share to and click "OK".
+
+The system will call Docker registry API to grant access for the user shared to after these operations. The model market will save the grant information in a database table `image_shares` that have below columns:
+
+1. `OwnerID`: Docker image owner user ID.
+1. `SharedUserID`: User ID that the image is shared to.
+
+## Steps to Publish a Trained Model
+
+The following SQL statement will publish a trained model named `my_first_model` to the model market.
+
+```sql
+SQLFLOW PUBLISH my_first_model
+    [TO https://models.sqlflow.org/user_name]
+```
+
+By publishing a trained model, the model market will save the trained model weights together with ownership information into two database tables (can use a MySQL service in general): the **trained models table** and the **evaluation result table**.
+
+### Trained Models Table
+
+Once a training job completes, the submitter program adds/updates a row of the trained models' table, which contains (at least) the following fields.
```

Comment:
The table size may increase greatly if the submitter adds a row for each training job. Can we set a lifecycle for each trained model record? The table will drop the trained model record after the lifecycle if the owner dese not public or share the trained model.

Reply:
The trained model record will only be added when a user executes `SQLFLOW PUBLISH ...` to publish a model. Publishing a model is done only when the trained model can be reused on other datasets or and the trained model is already getting pretty good performance (accuracy). So trained models in the list will not be too much and users can delete the model if the model is no longer needed.